onnxruntime/core/optimizer/qdq_transformer/qdq_final_cleanup.cc

```diff
@@ -31,98 +31,105 @@
+        graph.RemoveEdge(src_node_idx, first_node.Index(), src_arg_idx, 0);
+      }
+    }
 
-  // remove edge between pair we're removing
-  // both DQ and Q are single input single output so src idx and dest idx must be 0
-  graph.RemoveEdge(first_node.Index(), second_node.Index(), 0, 0);
+    // remove edge between pair we're removing
+    // both DQ and Q are single input single output so src idx and dest idx must be 0
+    graph.RemoveEdge(first_node.Index(), second_node.Index(), 0, 0);
 
-  if (!produces_graph_output) {
-    // remove edge to downstream node
-    const Node::EdgeEnd& output_edge = *second_node.OutputEdgesBegin();
-    downstream_node_idx = output_edge.GetNode().Index();
-    downstream_arg_idx = output_edge.GetDstArgIndex();
+    if (!produces_graph_output) {
+      // remove edge to downstream node
+      const Node::EdgeEnd& output_edge = *second_node.OutputEdgesBegin();
+      downstream_node_idx = output_edge.GetNode().Index();
+      downstream_arg_idx = output_edge.GetDstArgIndex();
 
-    // source arg idx is 0 as Q/DQ only has one output
-    graph.RemoveEdge(second_node.Index(), downstream_node_idx, 0, downstream_arg_idx);
+      // source arg idx is 0 as Q/DQ only has one output
+      graph.RemoveEdge(second_node.Index(), downstream_node_idx, 0, downstream_arg_idx);
 
-    // replace input on downstream node
-    Node& downstream_node = *graph.GetNode(downstream_node_idx);
-    downstream_node.MutableInputDefs()[downstream_arg_idx] = first_node.MutableInputDefs()[0];
+      // replace input on downstream node
+      Node& downstream_node = *graph.GetNode(downstream_node_idx);
+      downstream_node.MutableInputDefs()[downstream_arg_idx] = first_node.MutableInputDefs()[0];
 
-    // create edge between src_node (if available) and downstream node
-    if (input_edge) {
-      graph.AddEdge(src_node_idx, downstream_node_idx, src_arg_idx, downstream_arg_idx);
-    }
-  } else {
-    NodeArg* graph_output_nodearg = second_node.MutableOutputDefs()[0];
-    if (src_arg_idx >= 0) {
-      // update the src node to produce the graph output that was being provided by second_node
-      Node& src_node = *graph.GetNode(src_node_idx);
-      src_node.MutableOutputDefs()[src_arg_idx] = graph_output_nodearg;
+      // create edge between src_node (if available) and downstream node
+      if (input_edge) {
+        graph.AddEdge(src_node_idx, downstream_node_idx, src_arg_idx, downstream_arg_idx);
+      }
     } else {
-      // add Identity node to connect the graph input or initializer to the graph output.
-      Node& id_node = graph.AddNode(graph.GenerateNodeName("QDQFinalCleanupTransformer"),
-                                    "Identity", "", {first_node.MutableInputDefs()[0]}, {graph_output_nodearg});
-      id_node.SetExecutionProviderType(second_node.GetExecutionProviderType());
+      NodeArg* graph_output_nodearg = second_node.MutableOutputDefs()[0];
+      if (src_arg_idx >= 0 && second_node_ptrs.size() == 1) {
```

Comment:
there a corner case: Q->{DQ1->output1, DQ2->output2}. there will be 2 different graph output names. It is quite rare though.

Reply:
The `QDQFinalCleanupTransformer_BasicQDQsCleanup` covers this when the parameter  `is_dq_output = True`.